sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSources/LogAnalyticsDataFeedSource.cs

```diff
@@ -0,0 +1,106 @@
+        /// </summary>
+        /// <param name="workspaceId">The workspace ID of the Log Analytics resource.</param>
+        /// <param name="query">The query to retrieve the data to be ingested.</param>
+        /// <param name="clientId">The client ID used for AAD authentication.</param>
+        /// <param name="clientSecret">The client secret used for AAD authentication.</param>
+        /// <param name="tenantId">The tenant ID used for AAD authentication.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="workspaceId"/>, <paramref name="query"/>, <paramref name="clientId"/>, <paramref name="clientSecret"/>, or <paramref name="tenantId"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="workspaceId"/>, <paramref name="query"/>, <paramref name="clientId"/>, <paramref name="clientSecret"/>, or <paramref name="tenantId"/> is empty.</exception>
+        public LogAnalyticsDataFeedSource(string workspaceId, string query, string clientId, string clientSecret, string tenantId)
+            : base(DataFeedSourceType.LogAnalytics)
+        {
+            Argument.AssertNotNullOrEmpty(workspaceId, nameof(workspaceId));
+            Argument.AssertNotNullOrEmpty(query, nameof(query));
+            Argument.AssertNotNullOrEmpty(clientId, nameof(clientId));
+            Argument.AssertNotNullOrEmpty(clientSecret, nameof(clientSecret));
+            Argument.AssertNotNullOrEmpty(tenantId, nameof(tenantId));
+
+            WorkspaceId = workspaceId;
+            Query = query;
+            ClientId = clientId;
+            ClientSecret = clientSecret;
+            TenantId = tenantId;
+        }
+
+        internal LogAnalyticsDataFeedSource(AzureLogAnalyticsParameter parameter)
+            : base(DataFeedSourceType.LogAnalytics)
+        {
+            Argument.AssertNotNull(parameter, nameof(parameter));
+
+            WorkspaceId = parameter.WorkspaceId;
+            Query = parameter.Query;
+            ClientId = parameter.ClientId;
+            ClientSecret = parameter.ClientSecret;
+            TenantId = parameter.TenantId;
+        }
+
+        /// <summary>
+        /// The workspace ID of the Log Analytics resource.
+        /// </summary>
+        public string WorkspaceId { get; set; }
+
+        /// <summary>
+        /// The query to retrieve the data to be ingested.
+        /// </summary>
+        public string Query { get; set; }
+
+        /// <summary>
+        /// The client ID used for AAD authentication.
+        /// </summary>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        /// The tenant ID used for AAD authentication.
+        /// </summary>
+        public string TenantId { get; set; }
+
+        /// <summary>
+        /// The client secret used for AAD authentication.
+        /// </summary>
+        internal string ClientSecret { get; set; }
```

Comment:
should have the `Volatile` stuff to make sure it is thread-safe, right?

Reply:
You're right. Totally missed it. Updating it.

Reply:
Done.